book/src/dyn_tutorial/permissions.md

```diff
@@ -0,0 +1,126 @@
+| `my`         | Unique ownership                   | Nobody else has access to this object unless we grant it to them.                                                                                                                                                            |
+| `our`        | [Shared ownership](./share.md)     | Many variables can share access at once, but [none of them can write to the object](./sharing_xor_mutation.md).                                                                                                              |
+| `leased`     | [Unique lease](./lease.md)         | the object is [leased](./lease.md) from another variable. This variable has unique access, but it is only temporary. The owner can cancel the lease and start using the object again.                                        |
+| `our leased` | [Shared lease](./shared_leases.md) | a [shared lease](./shared_leases.md) means that multiple variables have ([read-only](./sharing_xor_mutation.md)) access, but that access is only temporary. The owner can cancel the lease and start using the object again. |
+| `any`        |                                    | the variable can store values with any of the above permisions                                                                                                                                                               |
+
+## Shared ownership
+
+For example, instead of writing `my p`, we could've written `our p` to declare a variable with shared ownership:
+
+```
+class Point(x, y)
+
+async fn main() {
+    our p = Point(22, 44)
+}
+```
+
+
+## Enforcing uniqueness and ownership
+
+When you declare a variable as `my p`, that means that it has *unique ownership* of the object(s) stored inside of it. 
+
+```
+async fn main() {
+    our p = Point(22, 44)
+    my q = p                                         # Error
+}
+```
+
+
+
+## Permissions and mutation
+
+The most obvious effect 
+## Default permissions
+
+If you don't specify any ownership permision, the default is `our leased`. This means that the variable cannot be used to mutate the fields of the object. Example:
+
+```
+class Point(x, y)
+
+async fn main() {
+    p = Point(22, 44)      # declares `p` in `our leased` mode
+    p.x += 1               # Error!
+}
+```
+
+## Ownership permissions are enforced
+
+We saw that `my p` indicates that the variable `p` has unique ownership of its contents. But what happens if we try to store a value in `p` that cannot be owned? As an example, integers like `22` and `44` are inherently shared. If we try to claim ownership of those, we get an error. As an example, try running this:
+
+```
+async fn main() {
+    my x = 22               # Error!
```

Comment:
What's the syntax to take ownership of a `22` here?

Reply:
Maybe it's just not possible? I can create an `our` and reassign the `our` as a way to "mutate" an integer, but I can't ever have a `my Int`?

Reply:
Yes. You can do `our x = 22`, which then permits you to do (e.g.) `x += 1`, but the "fields of the integer" are not changed. Think of like Java's `Integer` class.

Reply:
This suggests to me there are no immutable bindings in dada? Is that correct?

Thanks for all the clarifications.